src/common/radix_partitioning.cpp

```diff
@@ -0,0 +1,454 @@
+	case 7:
+		return OP::template Operation<7>(std::forward<ARGS>(args)...);
+	case 8:
+		return OP::template Operation<8>(std::forward<ARGS>(args)...);
+	case 9:
+		return OP::template Operation<9>(std::forward<ARGS>(args)...);
+	case 10:
+		return OP::template Operation<10>(std::forward<ARGS>(args)...);
+	default:
+		throw InternalException("TODO");
+	}
+}
+
+template <class OP, class RETURN_TYPE, idx_t radix_bits_1, typename... ARGS>
+RETURN_TYPE DoubleRadixBitsSwitch2(idx_t radix_bits_2, ARGS &&...args) {
+	D_ASSERT(radix_bits_2 <= sizeof(hash_t) * 8);
+	switch (radix_bits_2) {
+	case 1:
+		return OP::template Operation<radix_bits_1, 1>(std::forward<ARGS>(args)...);
+	case 2:
+		return OP::template Operation<radix_bits_1, 2>(std::forward<ARGS>(args)...);
+	case 3:
+		return OP::template Operation<radix_bits_1, 3>(std::forward<ARGS>(args)...);
+	case 4:
+		return OP::template Operation<radix_bits_1, 4>(std::forward<ARGS>(args)...);
+	case 5:
+		return OP::template Operation<radix_bits_1, 5>(std::forward<ARGS>(args)...);
+	case 6:
+		return OP::template Operation<radix_bits_1, 6>(std::forward<ARGS>(args)...);
+	case 7:
+		return OP::template Operation<radix_bits_1, 7>(std::forward<ARGS>(args)...);
+	case 8:
+		return OP::template Operation<radix_bits_1, 8>(std::forward<ARGS>(args)...);
+	case 9:
+		return OP::template Operation<radix_bits_1, 9>(std::forward<ARGS>(args)...);
+	case 10:
+		return OP::template Operation<radix_bits_1, 10>(std::forward<ARGS>(args)...);
+	default:
+		throw InternalException("TODO");
+	}
+}
+
+template <class OP, class RETURN_TYPE, typename... ARGS>
+RETURN_TYPE DoubleRadixBitsSwitch1(idx_t radix_bits_1, idx_t radix_bits_2, ARGS &&...args) {
+	D_ASSERT(radix_bits_1 <= sizeof(hash_t) * 8);
+	switch (radix_bits_1) {
+	case 1:
+		return DoubleRadixBitsSwitch2<OP, RETURN_TYPE, 1>(radix_bits_2, std::forward<ARGS>(args)...);
+	case 2:
+		return DoubleRadixBitsSwitch2<OP, RETURN_TYPE, 2>(radix_bits_2, std::forward<ARGS>(args)...);
+	case 3:
+		return DoubleRadixBitsSwitch2<OP, RETURN_TYPE, 3>(radix_bits_2, std::forward<ARGS>(args)...);
+	case 4:
+		return DoubleRadixBitsSwitch2<OP, RETURN_TYPE, 4>(radix_bits_2, std::forward<ARGS>(args)...);
+	case 5:
+		return DoubleRadixBitsSwitch2<OP, RETURN_TYPE, 5>(radix_bits_2, std::forward<ARGS>(args)...);
+	case 6:
+		return DoubleRadixBitsSwitch2<OP, RETURN_TYPE, 6>(radix_bits_2, std::forward<ARGS>(args)...);
+	case 7:
+		return DoubleRadixBitsSwitch2<OP, RETURN_TYPE, 7>(radix_bits_2, std::forward<ARGS>(args)...);
+	case 8:
+		return DoubleRadixBitsSwitch2<OP, RETURN_TYPE, 8>(radix_bits_2, std::forward<ARGS>(args)...);
+	case 9:
+		return DoubleRadixBitsSwitch2<OP, RETURN_TYPE, 9>(radix_bits_2, std::forward<ARGS>(args)...);
+	case 10:
+		return DoubleRadixBitsSwitch2<OP, RETURN_TYPE, 10>(radix_bits_2, std::forward<ARGS>(args)...);
+	default:
+		throw InternalException("TODO");
+	}
+}
+
+unique_ptr<idx_t[]> RadixPartitioning::InitializeHistogram(idx_t radix_bits) {
```

Comment:
Out of curiosity, why do we use this idiom everywhere instead of 
```cpp
return vector<idx_t>((1 << radix_bits), 0);
```

Reply:
I started off with building a histogram, but decided that we do not need one for the external hash join. Havent touched this code in a while and it's unused so deleting it for now.